doc/source/cluster/kubernetes/user-guides/pod-command.md

```diff
@@ -1,18 +1,69 @@
+KubeRay generates a `ray start` command for each Ray Pod.
+Sometimes, you may want to execute certain commands either before or after the ray start command, or you may wish to define the container's command yourself.
+This document shows you how to do that.
 
-* (2) **After `ray start` (RayCluster is ready)**: As an example, you can launch a Ray serve deployment when the RayCluster is ready.
+## Part 1: Specify your custom container command without any generated `ray start` command
 
-## Current KubeRay operator behavior for container commands
-* The current behavior for container commands is not finalized, and **may be updated in the future**.
-* See [code](https://github.com/ray-project/kuberay/blob/47148921c7d14813aea26a7974abda7cf22bbc52/ray-operator/controllers/ray/common/pod.go#L301-L326) for more details.
+Starting with KubeRay v1.1.0, if users add the annotation `ray.io/overwrite-container-cmd: "true"` to a RayCluster, KubeRay respects the container `command` and `args` as provided by the users, without including any generated command.
 
-## Timing 1: Before `ray start`
-Currently, for timing (1), we can set the container's `Command` and `Args` in RayCluster specification to reach the goal.
+```yaml
+apiVersion: ray.io/v1
+kind: RayCluster
+metadata:
+  annotations:
+    # If this annotation is set to "true", KubeRay will respect the container `command` and `args`.
+    ray.io/overwrite-container-cmd: "true"
+  ...
+spec:
+  headGroupSpec:
+    rayStartParams: {}
+    # Pod template
+    template:
+      spec:
+        containers:
+        - name: ray-head
+          image: rayproject/ray:2.8.0
+          # Because the annotation "ray.io/overwrite-container-cmd" is set to "true",
+          # KubeRay will overwrite the generated container command with `command` and
+          # `args` in the following. Hence, you need to specify the `ulimit` command
+          # by yourself to avoid Ray scalability issues.
+          command: ["/bin/bash", "-lc", "--"]
+          # Starting from v1.1.0, KubeRay injects the environment variable `KUBERAY_GEN_RAY_START_CMD`
+          # into the Ray container. This variable can be used to retrieve the generated Ray start command.
+          # Note that this environment variable does not include the `ulimit` command.
+          args: ["echo head; ulimit -n 65536; $KUBERAY_GEN_RAY_START_CMD"]
+          ...
+```
+
+The preceding example YAML is a part of [ray-cluster.overwrite-command.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.overwrite-command.yaml).
+
+* `metadata.annotations.ray.io/overwrite-container-cmd: "true"`: This annotation tells KubeRay to respect the container `command` and `args` as provided by the users, without including any generated command.
+
+* `ulimit -n 65536`: This command is necessary to avoid Ray scalability issues caused by running out of file descriptors.
```

Comment:
I forgot (and the reader might be curious), why didn't we just make ulimit -n 65536 part of the KUBERAY_GEN_RAY_START_CMD? Is it because some users might want to delete the ulimit part?

Reply:
If there's an interesting explanation for the above question it might be helpful to write it in the doc, maybe here or at the line below where we say "Note that this environment variable doesn't include the `ulimit` command."

Reply:
> why didn't we just make ulimit -n 65536 part of the KUBERAY_GEN_RAY_START_CMD? Is it because some users might want to delete the ulimit part?

Some users want to run some custom commands between `ulimit` and `ray start`. https://github.com/ray-project/kuberay/issues/1560#issuecomment-1836965826 is an example.

Reply:
Ah right, I forgot. Makes sense